tests/test_sdsc.py

```diff
@@ -59,6 +59,14 @@ def test_highlighter():
     assert xml == "highlight <highlight>these two</highlight> words"
 
 
+def test_contextpatternlocations():
+    """checks the contextpatternlocations function"""
+    assert sdsc.contextpatternlocations(1, [1]) == [1]
+    assert sdsc.contextpatternlocations(1, [1, 2]) == [1, 2]
```

Comment:
Well using a list to solve this seemed the simplest way to get all the possible multiplications.

Of course, the fact that the factors are handed to another function is new. Maybe bools would work: `contextpatternlocations(locationxpath, before=False, after=True, fuzzymode=False)`

And then, you could create the list in-function or just not use a list. Whatever you like.